vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -370,19 +383,27 @@ def collect_metrics_async(self):
                         e,
                     )
 
-    def make_batch(self, mors, metric_ids, resource_type):
+    def make_batch(
+        self,
+        mors,  # type: Iterable[vim.ManagedEntity]
+        metric_ids,  # type: List[vim.PerformanceManager.MetricId]
+        resource_type,  # type: Type[vim.ManagedEntity]
+    ):  # type: (...) -> Generator[MorBatch, None, None]
         """Iterates over mor and generate batches with a fixed number of metrics to query.
         Querying multiple resource types in the same call is error prone if we query a cluster metric. Indeed,
         cluster metrics result in an unpredicatable number of internal metric queries which all count towards
         max_query_metrics. Therefore often collecting a single cluster metric can make the whole call to fail. That's
         why we should never batch cluster metrics with anything else.
         """
         # Safeguard, let's avoid collecting multiple resources in the same call
-        mors = [m for m in mors if isinstance(m, resource_type)]
+        mors_filtered = []  # type: List[vim.ManagedEntity]
+        for m in mors:
+            if isinstance(m, resource_type):  # type: ignore
+                mors_filtered.append(m)
```

Comment:
Looks like mypy is fine with the list comprehension (as it should):

```suggestion
        mors_filtered = [m for m in mors if isinstance(m, resource_type)]  # type: List[vim.ManagedEntity]
```

Reply:
Indeed, I got some errors previously, but it was probably due to something else.